hudi-flink-datasource/hudi-flink/src/test/java/org/apache/hudi/utils/TestFlinkWriteClients.java

```diff
@@ -55,7 +57,19 @@ void testAutoSetupLockProvider() throws Exception {
     conf.setBoolean(FlinkOptions.METADATA_ENABLED, true);
     StreamerUtil.initTableIfNotExists(conf);
     HoodieWriteConfig writeConfig = FlinkWriteClients.getHoodieClientConfig(conf, false, false);
-    assertThat(writeConfig.getLockProviderClass(), is(FileSystemBasedLockProvider.class.getName()));
+    Assertions.assertTrue(writeConfig.isLockRequired());
+    assertThat(writeConfig.getLockProviderClass(), is(InProcessLockProvider.class.getName()));
+    assertThat(writeConfig.getWriteConcurrencyMode(), is(WriteConcurrencyMode.SINGLE_WRITER));
+  }
+
+  @Test
+  void testAutoSetupLockProviderWithoutTableService() throws Exception {
+    conf.setBoolean(FlinkOptions.METADATA_ENABLED, true);
+    conf.setBoolean("hoodie.table.services.enabled", false);
+    StreamerUtil.initTableIfNotExists(conf);
```

Comment:
We can not use in-progress lock for Flink, it does not work well in distributed env.

Reply:
Can you explain why it doesn't work well?

Reply:
Because the instant in Flink does not always commit through the coordinator, some are committed through sink operator.

Reply:
Ok, so hudi table based on object storage can't enable metadata table directly, because most of them can not support atomic creation

Reply:
we have file based lock provider instead.

Reply:
> we have file based lock provider instead.

As I said earlier, object storage not support atomic creation, file based lock is invalid, see https://github.com/apache/hudi/pull/7614